### PR TITLE
move golangci-lint to config file

### PR DESCRIPTION
**Pull request by binjip978**

### CODE REVIEW

**Comment by binjip978** [comment]
now it can be run locally
```
go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.56.0
GOOS=linux GOARCH=amd64 golangci-lint run
```

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```
Comment (hugelgupf) [inline .github/workflows/golangci-lint.yml:6]: add pull request?
also needs

```
pull_request:
  branches:
    - main
```

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+permissions:
+  contents: read
```
Comment (hugelgupf) [inline .github/workflows/golangci-lint.yml:9]: pull-requests: read

File: .github/workflows/golangci-lint.yml (hunk 1 above)
Comment (binjip978) [reply .github/workflows/golangci-lint.yml:6]: I think it works even without that, but added anyway.

File: .github/workflows/golangci-lint.yml (hunk 2 above)
Comment (binjip978) [reply .github/workflows/golangci-lint.yml:11]: added

**Review by RiSKeD — COMMENTED** [review]
nice changes. It might break the scope of this PR, but it would be nice to enable more linters. It really depends on the amount of refactoring necessary.

**Review by RiSKeD — APPROVED** [review]

**Comment by binjip978** [comment]
> nice changes. It might break the scope of this PR, but it would be nice to enable more linters. It really depends on the amount of refactoring necessary.

yes some of the checks will be easy to fix, but probably it will be better to do in next PR.

**Pull request merged**